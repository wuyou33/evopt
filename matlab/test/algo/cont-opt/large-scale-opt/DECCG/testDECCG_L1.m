clear; clc;

% NOTE that put this code in a folder named *deccg*, which could be
% downloaded from * http://staff.ustc.edu.cn/~ketang/codes/DECCG.zip *.

%%
rng('shuffle'); % for uncontrollable randomness

addpath('benchmark');

funcDim = 1000;
numSubDim = 100;
groupNum = 10;
popSize = 100;
testNum = 25;
funcEvalMax = 5000 * funcDim;
genMax = funcEvalMax / popSize;

isGreedyAfterCC = true;
% isGreedyAfterCC = false;
isAdaptiveWeight = true;
% isAdaptiveWeight = false;

optResFilename = 'result/testDECCG_f%02d.mat';

global initial_flag;
for f = 1 : 13
    optys = [];
    optxs = [];
    funcEvalCurves = cell(1, testNum);
    if f == 1 || f == 3 || f == 4 || f == 6
        funcLowerBounds = -100 * ones(popSize, funcDim);
        funcUpperBounds = 100 * ones(popSize, funcDim);
    elseif f == 2
        funcLowerBounds = -10 * ones(popSize, funcDim);
        funcUpperBounds = 10 * ones(popSize, funcDim);
    elseif f == 5
        funcLowerBounds = -30 * ones(popSize, funcDim);
        funcUpperBounds = 30 * ones(popSize, funcDim);
    elseif f == 7
        funcLowerBounds = -1.28 * ones(popSize, funcDim);
        funcUpperBounds = 1.28 * ones(popSize, funcDim);
    elseif f == 8
        funcLowerBounds = -500 * ones(popSize, funcDim);
        funcUpperBounds = 500 * ones(popSize, funcDim);
    elseif f == 9
        funcLowerBounds = -5.12 * ones(popSize, funcDim);
        funcUpperBounds = 5.12 * ones(popSize, funcDim);
    elseif f == 10
        funcLowerBounds = -32 * ones(popSize, funcDim);
        funcUpperBounds = 32 * ones(popSize, funcDim);
    elseif f == 11
        funcLowerBounds = -600 * ones(popSize, funcDim);
        funcUpperBounds = 600 * ones(popSize, funcDim);
    elseif f == 12 || f == 13
        funcLowerBounds = -50 * ones(popSize, funcDim);
        funcUpperBounds = 50 * ones(popSize, funcDim);
    end
    for t = 1 : testNum
        initial_flag = 0;
        [opty, optx, funcEvalCurve, optxSeq, optxFlag] = ...
            lsoDECCG_L1(...
            'benchmark_func', f, funcDim, ...
            funcLowerBounds, funcUpperBounds, ...
            popSize, genMax, numSubDim, ...
            isAdaptiveWeight, isGreedyAfterCC);
        optys = cat(1, optys, opty);
        optxs = cat(1, optxs, optx);
        funcEvalCurves{t} = cat(1, funcEvalCurves{t}, funcEvalCurve);
    end
    save(sprintf(optResFilename, f), ...
        'optys', 'optxs', 'funcEvalCurves', 'optxFlag');
    fprintf(sprintf('f%02d: %7.5e (%7.5e)\n', ...
        f, mean(optys), std(optys)));
end
