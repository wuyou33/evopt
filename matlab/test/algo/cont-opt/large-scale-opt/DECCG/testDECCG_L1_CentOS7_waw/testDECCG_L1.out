
                            < M A T L A B (R) >
                  Copyright 1984-2016 The MathWorks, Inc.
                   R2016b (9.1.0.441655) 64-bit (glnxa64)
                             September 7, 2016

f01: 7.32870e-32 (2.39676e-32)
f02:     Inf (    NaN)
f03: 2.50842e+05 (1.76300e+04)
f04: 8.94175e-05 (2.78021e-05)
f05: 9.85730e+02 (3.63545e-01)
f06: 2.16000e+00 (2.30362e+00)
f07: 2.88136e-01 (1.67738e-02)
f08: -4.18983e+05 (2.00936e-10)
f09: 1.72111e+02 (8.03828e+01)
f10: 1.03421e-07 (5.94126e-08)
f11: 1.33795e-13 (6.63934e-13)
f12: 6.31888e-04 (1.29025e-03)
f13: 6.35244e-27 (1.09526e-26)
